source/common/protobuf/utility.cc

```diff
@@ -197,12 +198,7 @@ void deprecatedFieldHelper(Runtime::Loader* runtime, bool proto_annotated_as_dep
   if (warn_only) {
     ENVOY_LOG_MISC(warn, "{}", with_overridden);
   } else {
-    const char fatal_error[] =
-        " If continued use of this field is absolutely necessary, see "
-        "https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime"
-        "#using-runtime-overrides-for-deprecated-features for how to apply a temporary and "
-        "highly discouraged override.";
-    throw ProtoValidationException(with_overridden + fatal_error, message);
```

Comment:
The validation visitors above take this exception and decide to warn/throw depending on whether they are strict/warning visitors. The validation visitors are configured on CLI to reflect the preferences wrt unknown field handling. I think in both cases, for deprecated, we want them to throw, since the code path that led us here indicates that we must do so.

CC @alyssawilk

Reply:
@htuch since the action (log, throw exception) that validator visitor needs to take does not depend on runtime type but on code logic, i could think of multiple (not pretty) approaches how it could be handled:
1. No need to pass visitor into method, invoke static method of corresponding visitor type (`StrictValidationVisitorImpl::onDeprecatedField`) , which involves changing signature of method to static.
2. Share same implementation between 2 visitor types for `onDeprecatedField` method, the implementation would throw an exception. This would be slightly semantically misleading, as warning visitor would throw an exception.

Reply:
Unless you have other suggestion, second one sounds like less evil.

Reply:
I think (2) sounds good. I think if we were being pedantic, we would rename the visitors, the warning visitor would be the `UnknownFieldWarningVisitor`, etc.